src/python/pants/core/goals/fmt.py

```diff
@@ -204,19 +239,27 @@ async def fmt(
         merged_formatted_digest = await Get(Digest, MergeDigests(changed_digests))
         workspace.write_digest(merged_formatted_digest)
 
-    sorted_results = sorted(individual_results, key=lambda res: res.formatter_name)
-    for result in sorted_results:
-        console.print_stderr(
-            f"{console.green('✓')} {result.formatter_name} made no changes."
-            if not result.did_change
-            else f"{console.red('𐄂')} {result.formatter_name} made changes."
-        )
-        if result.stdout:
-            console.print_stderr(result.stdout)
-        if result.stderr:
-            console.print_stderr(result.stderr)
-        if result != sorted_results[-1]:
-            console.print_stderr("")
+    if individual_results:
+        console.print_stderr("")
+
+    # We group all results for the same formatter so that we can give one final status in the
+    # summary. This is only relevant if there were multiple results because of
+    # `--per-target-caching`.
+    formatter_to_results = defaultdict(set)
+    for result in individual_results:
+        formatter_to_results[result.formatter_name].add(result)
+
+    for formatter, results in sorted(formatter_to_results.items()):
+        if any(result.did_change for result in results):
+            sigil = console.red("𐄂")
+            status = "made changes"
+        elif all(result.skipped for result in results):
```

Comment:
To confirm my understanding, `any` would also work here since you can only skip a formatter or not globally and not per individual operand.

Reply:
Yes, that is correct. I figured that `all()` is more defensive, though. It's up to the individual formatter to determine how it wants to skip things. Right now, each Python formatter has an option `--skip`. They technically could use some other scheme for skipping, like "skip if the file is `forbidden.py`"

Reply:
OK. When that defense eventually gets utilized it will probably require circling back and handling the dropped case separately instead of falling into green with something like `green (N operands skipped)`.